### PR TITLE
Documented remaining values

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Changed a lot of types & functions from exported to unexported.

- Documented the remaining exported types that needs to be exported, such as JSON responses and DB models.

## Motivation

We start growing error fatigue due to the existing errors. Broken window principle applies here.

If we have properly commented everything, it'll be easier to see how we should keep it, and it's be less insentive to just ignore the linting errors.

### CODE REVIEW

**Comment by Alexamakans** [comment]
Nice!

I noticed that even if the type is unexported swaggo still generates a file for it, so e.g. `TestsResults` could be `testsResults`.
I seemed to have no issues when doing that, at least.

**Comment by jilleJr** [comment]
> Nice!
> 
> I noticed that swaggo still generates a file for unexported types as well, so e.g. `TestsResults` could be `testsResults`.
> I seemed to have no issues when doing that, at least.

Well I'll file that under "better to have them documented than not", as I find it quite nice to have the comments in the swagger spec, ex:

Screenshot from 2021-08-20 15-30-07

**Comment by Alexamakans** [comment]
> > Nice!
> > I noticed that swaggo still generates a file for unexported types as well, so e.g. `TestsResults` could be `testsResults`.
> > I seemed to have no issues when doing that, at least.
> 
> Well I'll file that under "better to have them documented than not", as I find it quite nice to have the comments in the swagger spec, ex:
> 
> Screenshot from 2021-08-20 15-30-07

Ah! Didn't know it showed up there. Nice.

File: database_models.go [hunk 1]
```diff
@@ -129,6 +142,7 @@ type Log struct {
 	Timestamp time.Time `gorm:"not null" json:"timestamp" format:"date-time"`
 }
 
+// Param holds the definition of a input parameter for a project.
```
Comment (Alexamakans) [inline database_models.go:145]: typo
```suggestion
// Param holds the definition of an input parameter for a project.
```

File: project.go [hunk 2]
```diff
@@ -23,22 +23,25 @@ import (
...
 )
 
-type ProjectModule struct {
+type projectModule struct {
 	Database     *gorm.DB
 	MessageQueue *messagebus.MQConnection
 	Config       *Config
 }
 
+// PaginatedBuilds is a list of builds as well as an explicit total cound field.
```
Comment (Alexamakans) [inline project.go:32]: typo
```suggestion
// PaginatedBuilds is a list of builds as well as an explicit total count field.
```

**Review by Alexamakans — APPROVED** [review]
2 typos, rest LGTM!

**Review by iverestefans — APPROVED** [review]
LGTM!

**Pull request merged**